tests/src/test/scala/com/nvidia/spark/rapids/HashAggregatesSuite.scala

```diff
@@ -1647,44 +1540,40 @@ class HashAggregatesSuite extends SparkQueryCompareTestSuite {
       res
   }
 
-  testSparkResultsAreEqual("PartMerge:avg_overflow_cast_dbl", veryLargeLongsFromCSVDf,
+  testSparkResultsAreEqualWithCapture("PartMerge:avg_overflow_cast_dbl", veryLargeLongsFromCSVDf,
     conf = floatAggConf, repart = 2) {
-    frame => val result = frame.groupBy("large_longs").agg(avg("large_longs"))
-      checkExecNode(result)
-      result
-  }
+    frame => frame.groupBy("large_longs").agg(avg("large_longs"))
+  } { (_, gpuPlan) => checkExecPlan(gpuPlan) }
 
-  ALLOW_NON_GPU_testSparkResultsAreEqual(
+  ALLOW_NON_GPU_testSparkResultsAreEqualWithCapture(
     "max_with_nans_fall_back",
     nanDf,
     Seq("HashAggregateExec", "AggregateExpression",
       "AttributeReference", "Alias", "Max"),
     conf = new SparkConf()) {
-    frame => val result = frame.agg(max("doubles"))
-      // verify nothing ran on the gpu
-      if (result.queryExecution.executedPlan.conf.getAllConfs(
-        RapidsConf.SQL_ENABLED.key).toBoolean) {
-        val execNode = result.queryExecution.executedPlan.find(_.isInstanceOf[GpuHashAggregateExec])
-        assert(!execNode.isDefined)
-      }
-      result
-  }
+    frame => frame.agg(max("doubles"))
+  } { (_, gpuPlan) => {
+    // verify nothing ran on the gpu
+    if (gpuPlan.conf.getAllConfs(RapidsConf.SQL_ENABLED.key).toBoolean) {
+      val execNode = gpuPlan.find(_.isInstanceOf[GpuHashAggregateExec])
+      assert(!execNode.isDefined)
+    }
+  }}
 
-  ALLOW_NON_GPU_testSparkResultsAreEqual(
+  ALLOW_NON_GPU_testSparkResultsAreEqualWithCapture(
     "min_with_nans_fall_back",
     nanDf,
     Seq("HashAggregateExec", "AggregateExpression",
       "AttributeReference", "Alias", "Min"),
     conf = new SparkConf()) {
-    frame => val result = frame.agg(min("doubles"))
-      // verify nothing ran on the gpu
-      if (result.queryExecution.executedPlan.conf.getAllConfs(
-        RapidsConf.SQL_ENABLED.key).toBoolean) {
-        val execNode = result.queryExecution.executedPlan.find(_.isInstanceOf[GpuHashAggregateExec])
-        assert(!execNode.isDefined)
-      }
-      result
-  }
+    frame => frame.agg(min("doubles"))
+  } { (_, gpuPlan) => {
```

Comment:
nit, does the first slot in this tuple get utilized? Seems like we could change the signature of  `checkExecPlan` to take the tuple also. Again just a nit.

Reply:
This test doesn't perform any validation that requires knowledge of how the query executed on the CPU, so doesn't utilize the first value in the tuple. Other tests may potentially want to examine the CPU plan as well? It seemed to make sense to make it available.

We could just pass the tuple, but I think it would be less readable seeing `_1` or `_2` when examining the plans. Another option would be to introduce a `case class CpuGpuResult(cpuResult: SparkPlan, gpuResult: SparkPlan)` or similar.

Reply:
> Other tests may potentially want to examine the CPU plan as well? It seemed to make sense to make it available.

Yes that makes sense. I wasn't sure if that was the purpose (for future tests). Thanks!